[DX-1726] Gateway release notes 5.3.8 (#5658)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md

---

## 5.3.8 Release Notes

### Release Date XXX

### Release Highlights

This release focuses mainly on bug fixes. For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.3.8">}}) below.

### Breaking Changes

> There are no breaking changes in this release, however if moving from an version of Tyk older than 5.3.0 please read the

Reviewer: Please follow the [RN](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#breaking-changes) format

explanation provided with [5.3.0 release]({{< ref "#TykOAS-v5.3.0">}}).

### Deprecations

> This is an advanced notice that External OAuth, OpenID Connect (OIDC) authentication methods, and SQLite support will be deprecated starting in version 5.7.0. Please review your API configurations, as the Gateway logs will provide notifications for any APIs utilizing these methods.

Reviewer: Please add relevant reference links to `External OAuth, OpenID Connect, Sqlite`

Reviewer: Can you incorporate suggestion from @andyo-tyk review in dashboard PR

### Upgrade Instructions

> If you are upgrading to 5.3.8, please follow the detailed [upgrade instructions](#upgrading-tyk).

Reviewer: Please follow the [RN](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#upgrading-tyk) format

### Downloads

- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.3.8)
  - ```bash
    docker pull tykio/tyk-gateway:v5.3.8
    ```
- Helm charts
  - [tyk-charts v2.0.0]({{<ref "product-stack/tyk-charts/release-notes/version-2.0.md">}})
- [Source code tarball for OSS projects](https://github.com/TykTechnologies/tyk/releases)

### Changelog {#Changelog-v5.3.8}

<!-- Required. The change log should include the following ordered set of sections below that briefly summarise the features, updates and fixed issues of the release.
Here it is important to explain the benefit of each changelog item. As mentioned by James in a previous Slack message (https://tyktech.slack.com/archives/C044R3ZTN6L/p1686812207060839?thread_ts=1686762128.651249&cid=C044R3ZTN6L):
"...it is important to document the customer impact for the work delivered, so we can share it with prospects/install base. For example:
"New Chart delivers x and y benefit to a and b customer use cases. The business impact for them will be this and that" -->

#### Fixed

<ul>
<li>
<details>
<summary>Memory consumption reduced in Gateway for large payloads</summary>

> This update fixes a bug that caused increased memory usage when proxying large response payloads in version 5.3.1, restoring memory requirements to the levels seen in version 5.0.6. Users experiencing out-of-memory errors with 1GB+ file downloads will notice improved performance and reduced latency.

Reviewer: Does this mean that memory issue was only present in version `5.3.1` or all the versions after `5.3.1`?

Please rephrase if incorrect.

Author: Only present in 5.3.1

</details>
</li>
<li>
<details>
<summary>Path-Based Permissions in Combined Policies Not Preserved</summary>

We resolved an issue that caused path-based permissions in policies to be lost when policies were combined, potentially omitting URL values and restricting access based on the merge order. It ensures that all applicable policies merge their allowed URL access rights, regardless of the order in which they are applied.
</details>
</li>
<li>
<details>
<summary>Enhanced flexibility in OAS JSON Schema: Additional Properties no longer enforced</summary>

> A backwards compatibility issue in the OAS API Definitions JSON schema has been addressed by reinstating support for 'additionalProperties'. Previously, version 5.3 enforced strict validation, causing problems for users downgrading from version 5.6. With this change, Tyk customers can move between versions seamlessly, ensuring their APIs remain functional and avoiding system performance issues.

Reviewer: ```suggestion
A backwards compatibility issue in the OAS API Definitions JSON schema has been addressed by reinstating support for `additionalProperties`. Previously, version 5.3 enforced strict validation, causing problems for users downgrading from version 5.6. With this change, Tyk customers can move between versions seamlessly, ensuring their APIs remain functional and avoiding system performance issues.
```

</details>
</li>
<li>
<details>

> <summary>Deprecation notice for External OAuth and OpenID Connect options</summary>

Reviewer: I think this should not come in the `fixed` section. Used [added](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#added) instead.

> A deprecation notice for External OAuth and OpenID Connect (OIDC) authentication mechanisms has been added to the Gateway logs starting from version 5.3.8. This provides advanced notification to users regarding any APIs configured with these authentication methods in preparation for future upgrades where these middleware options may be removed in version 5.7.0.

Reviewer: ```suggestion
A [deprecation notice](#<add-deprecation-notice>) for External OAuth and OpenID Connect (OIDC) authentication mechanisms has been added to the Gateway logs starting from version 5.3.8. This provides advanced notification to users regarding any APIs configured with these authentication methods in preparation for future upgrades where these middleware options may be removed in version 5.7.0.
```

Add deprecation notice fragment

Author: The notice hasn't been posted anywhere in particular. However, we have added warning messages to all the pages that mention both authentication in Tyk docs.

</details>
</li>

<li>
<details>
<summary>Fix for API Key Loss on Worker Gateways Due to Keyspace Sync Interruption</summary>

> This update resolves an issue where API keys were lost on worker gateways if the keyspace synchronization process was interrupted. The solution now enforces a resynchronization whenever a connection is re-established between MDCB and a Worker Gateway, ensuring key data integrity and seamless API access.

Reviewer: Can you add reference documentation that elaborates more on `keyspace synchronization` feature? Ignore if not available.

Author: @caroltyk Please can you help here.

Reviewer 2: ```suggestion
This update resolves an issue where API keys were lost on worker gateways if the [keyspace synchronization]({{<ref "product-stack/tyk-enterprise-mdcb/advanced-configurations/synchroniser">}}) process was interrupted. The solution now enforces a resynchronization whenever a connection is re-established between MDCB and a Worker Gateway, ensuring key data integrity and seamless API access.
```

---

## 5.3.8 Release Notes

> ### Release Date XXX

Reviewer: ```suggestion
### Release Date 07 November 2024
```

#### 3rd Party Dependencies & Tools

<!-- Required. Third-party dependencies encompass tools (GoLang, Helm etc.), databases (PostgreSQL, MongoDB etc.) and external software libraries. This section should be a table that presents the third-party dependencies and tools compatible with the release. Compatible is used in the sense of those versions tested with the releases. Such information assists customers considering upgrading to a specific release.

Additionally, a disclaimer statement was added below the table, for customers to check that the third-party dependency they decide to install remains in support.

An example is given below for illustrative purposes only. Tested Versions and Compatible Versions information will require discussion with relevant squads and QA. -->

| Third Party Dependency                                        | Tested Versions       | Compatible Versions   | Comments                                                                                   |
| ------------------------------------------------------------- | --------------------- | --------------------- | ------------------------------------------------------------------------------------------ |

> | [Go](https://go.dev/dl/)                                      | 1.19 (GQL), 1.22 (GW) | 1.19 (GQL), 1.22 (GW) | [Go plugins]({{< ref "plugins/supported-languages/golang" >}}) must be built using Go 1.22 |

Reviewer: Does `1.19 (GQL)` has to be removed? It is not present in previous release.

</details>
</li>
</ul>

#### Fixed

<ul>
<li>
<details>
<summary>Memory consumption reduced in Gateway for large payloads</summary>

This update fixes a bug that caused increased memory usage when proxying large response payloads in version 5.3.1, restoring memory requirements to the levels seen in version 5.0.6. Users experiencing out-of-memory errors with 1GB+ file downloads will notice improved performance and reduced latency.
</details>
</li>
<li>
<details>
<summary>Path-Based Permissions in Combined Policies Not Preserved</summary>

We resolved an issue that caused path-based permissions in policies to be lost when policies were combined, potentially omitting URL values and restricting access based on the merge order. It ensures that all applicable policies merge their allowed URL access rights, regardless of the order in which they are applied.
</details>
</li>
<li>
<details>

> <summary>Enhanced flexibility in OAS JSON Schema: Additional Properties no longer enforced</summary>

Reviewer 3: ```suggestion
<summary>Enhanced flexibility in Tyk OAS schema validation</summary>
```

> A backwards compatibility issue in the OAS API Definitions JSON schema has been addressed by reinstating support for `additionalProperties`. Previously, version 5.3 enforced strict validation, causing problems for users downgrading from version 5.6. With this change, Tyk customers can move between versions seamlessly, ensuring their APIs remain functional and avoiding system performance issues.

Reviewer 3: ```suggestion
A backwards compatibility issue in the way that the Gateway handles Tyk OAS API definitions has been addressed by reducing the strictness of validation against the expected schema. Since Tyk version 5.3, the Gateway has enforced strict validation, potentially causing problems for users downgrading from newer versions. With this change, Tyk customers can move between versions seamlessly, ensuring their APIs remain functional and avoiding system performance issues.
```
What about this?

</details>
</li>
<li>
<details>
<summary>Fix for API Key Loss on Worker Gateways Due to Keyspace Sync Interruption</summary>

> This update resolves an issue where API keys were lost on worker gateways if the [keyspace synchronization]({{<ref "product-stack/tyk-enterprise-mdcb/advanced-configurations/synchroniser">}}) process was interrupted. The solution now enforces a resynchronization whenever a connection is re-established between MDCB and a Worker Gateway, ensuring key data integrity and seamless API access.

Reviewer 3: ```suggestion
This update resolves an issue where API keys could be lost if the [keyspace synchronization]({{<ref "product-stack/tyk-enterprise-mdcb/advanced-configurations/synchroniser">}}) between control and data planes was interrupted. The solution now enforces a resynchronization whenever a connection is re-established between MDCB and the data plane, ensuring key data integrity and seamless API access.
```
Please check with Platform squad, but I think this is perhaps better, since we are being consistent with control/data plane terminology rather than highlighting individual gateways.
